src/Chapter2.hs

```diff
@@ -348,9 +350,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf = take =<< halfLen
```

Comment:
Wow, very interesting solution!
Nice usage of the `Monad` instance for the arrow 🧠 🤯

Reply:
Thanks 😄 
Had do some reading on the bind and reverse bind arrow to understand what it was doing, and then I was so happy that it worked 😆